workers/gossip-server/src/peer_discovery/peers.rs

```diff
@@ -102,11 +109,19 @@ impl GossipProtocolExecutor {
                 .collect_vec()
         }; // locked_expiry_cache released
 
+        let my_cluster_id = self.global_state.get_cluster_id()?;
+        let num_local_peers =
+            filtered_peers.iter().filter(|peer| peer.cluster_id == my_cluster_id).count();
+        let num_remote_peers = filtered_peers.len() - num_local_peers;
+
         // Add all filtered peers to the network manager's address table
         self.add_new_addrs(&filtered_peers)?;
         // Add all filtered peers to the global peer index
         self.global_state.add_peer_batch(filtered_peers)?;
 
+        metrics::gauge!(NUM_LOCAL_PEERS_METRIC).increment(num_local_peers as f64);
+        metrics::gauge!(NUM_REMOTE_PEERS_METRIC).increment(num_remote_peers as f64);
```

Comment:
it looks like we never decrement this metric? We need to do so when peers expire o/w this metric only grows

Reply:
woah, strange - i definitely added a decrement at expiry, I must've lost in when rebasing or something. thank you for the call out!